### PR TITLE
Tags: adding, removing, updating and search

### PR DESCRIPTION
**Pull request by OleksandrYagelskiy**
- Allow adding notes with tags, update add-note command
- Add option to search notes by tag
- Update `change-note` command to allow optionally change note value or tags
- Improve dispaying tag
- Improve code and readme docs

### CODE REVIEW

File: console_bot/note.py [hunk 1]
```diff
@@ -83,7 +140,7 @@ def generate_random(self, save=True):
 
     def __str__(self):
         """Convert notebook to string."""
-        return "\n".join(f"{index}: {note}" for index, note in enumerate(sorted(self.data, key=lambda x: x.value)))
```
Comment (TetianaFilonenko) [inline console_bot/note.py:86]: @svilav FYI I had to remove sorting here as we rely on index in multiple places, like removal and changing notes
With default sorting, ordered by create time, we can easily rely on index, with sorting by value things get missed
I would rather have sorting as a separate command or option in all-notes command

File: README.md [hunk 2]
```diff
@@ -60,6 +62,7 @@ Console Bot supports the following commands:
 | `birthdays-for [days]`                          | Shows birthdays for all contacts celebrating in the next amount of days. |
 | `add-note`                                      | Adds note to Note Book.                                                  |
 | `find-notes`                                    | Search notes by keywords.                                                |
+| `find-notes-by-tag`                             | Search notes by tag.                                                     |
 | `delete-note`                                   | Delete note by index in Note Book.                                       |
 | `edit-note`                                     | Edit note by index in Note Book.                                         |
```
Comment (svilav) [inline README.md]:
```suggestion
| `change-note`                                     | Edit note by index in Note Book.                                         |
```

Comment (svilav) [reply README.md]: plus here is no description of `edit` command for AddressBook

File: console_bot/console_bot.py [hunk 3]
```diff
@@ -28,6 +28,7 @@
     "birthdays",
     "add-note",
     "find-notes",
+    "find-notes-by-tag",
```
Comment (svilav) [inline console_bot/console_bot.py:33]: `birthdays-for` is missing in this list

**Review by svilav — APPROVED** [review]
Everything seems to work. Thanks a lot!

**Pull request merged**